addons/stock/models/stock_picking.py

```diff
@@ -1004,11 +1004,12 @@ def _send_confirmation_email(self):
         subtype_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment')
         for stock_pick in self.filtered(lambda p: p.company_id.stock_move_email_validation and p.picking_type_id.code == 'outgoing'):
             delivery_template = stock_pick.company_id.stock_mail_confirmation_template_id
-            stock_pick.with_context(force_send=True).message_post_with_source(
-                delivery_template,
-                email_layout_xmlid='mail.mail_notification_light',
-                subtype_id=subtype_id,
-            )
+            if delivery_template:
```

Comment:
```suggestion
            if not delivery_template:
                continue
```
Apply minimal changes to ensure the original blame history remains intact in the stable branch.